src/azul/service/avro_pfb.py

```diff
@@ -153,24 +150,39 @@ def _add_missing_fields(cls, name: str, object_: MutableJSON, schema):
         None is the default value, but because of https://github.com/DataBiosphere/azul/issues/2370
         this isn't currently reflected in the schema.
         """
-        object_schema = one(f for f in schema['fields'] if f['name'] == 'object')
-        entity_schema = one(e for e in object_schema['type'] if e['name'] == name)
+        if schema['type'] == 'record':
+            object_schema = one(f for f in schema['fields'] if f['name'] == 'object')
+            entity_schema = one(e for e in object_schema['type'] if e['name'] == name)
+        elif schema['name'] == 'related_files':
+            entity_schema = schema['type']['items']
+        else:
+            assert False, schema
         for field in entity_schema['fields']:
             if field['name'] not in object_:
-                if name == 'files':
-                    default_value = None
+                if name == 'files' or name == 'related_files':
+                    if 'string' in field['type']:
+                        default_value = None
+                    elif field['type']['type'] == 'array':
+                        default_value = [None]
+                    else:
+                        assert False, field
```

Comment:
I think this might be just a little bit stricter than necessary. If `size` was left out for some reason, for example, we would have to change this code to accommodate it. I think `default_value` should be `None` for everything unless it's an array, or other composite type.

Reply:
All the `_nullable_to_pfb_types` include "string" which is why I thought this would be acceptable. e.g. `null_int: ['string', 'long']`.
With that said I've taken your comment and tried to make this section a bit more general as you suggested. The nature of the `field['type']` variable made this a little tricky though since I observed it either containing a list of types (`field['type'] == ['string']`) or a further nested dictionary (`field['type']['type'] == 'array'`)

Reply:
> All the _nullable_to_pfb_types include "string" which is why I thought this would be acceptable

You're totally right, I didn't think about this. What you wrote before is totally acceptable then. 

Possibly you already know this, but every type contains 'string' also because Terra doesn't allow `null`. Instead we represent it with an empty string. 

So the `'string'` you wrote here is the same as `null`. Consider adding a FIXME like `FIXME: Change 'string' to 'null' in https://github.com/DataBiosphere/azul/issues/2462`